File: LeetCode/2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.py
class Solution:
    def intersection(self, nums: List[List[int]]) -> List[int]:
        arr = [0] * 1001
        n = len(nums)
        for num in nums:
            for x in num:
                arr[x] += 1
        ans = []
        for i in range(0, 1001):
            if arr[i] == n:
                ans.append(i)
        return ans





# class Solution:
#     def intersection(self, nums: List[List[int]]) -> List[int]:
#         mp = {}
#         n = len(nums)
#         for num in nums:
#             for x in num:
#                 if x not in mp:
#                     mp[x] = 1
#                 else:
#                     mp[x] += 1
#         ans = []
#         for key, val in mp.items():
#             if val == n:
#                 ans.append(key)
#         ans.sort()
#         return ans

File: LeetCode/2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.java
class Solution {
    public List<Integer> intersection(int[][] nums) {
        int n = nums.length;
        int[] arr = new int[1001];
        for (int[] num : nums) {
            for (int x : num) {
                arr[x]++;
            }
        }
        List<Integer> ans = new ArrayList<>();
        for (int i = 0; i < 1001; i++) {
            if (arr[i] == n) {
                ans.add(i);
            }
        }
        return ans;
    }
}




// class Solution {
//     public List<Integer> intersection(int[][] nums) {
//         int n = nums.length;
//         Map<Integer, Integer> mp = new HashMap<>();
//         for (int[] num : nums) {
//             for (int x : num) {
//                 mp.put(x, mp.getOrDefault(x, 0) + 1);
//             }
//         }
//         List<Integer> ans = new ArrayList<>();
//         for (Map.Entry<Integer, Integer> entry : mp.entrySet()) {
//             if (entry.getValue() == n) {
//                 ans.add(entry.getKey());
//             }
//         }
//         Collections.sort(ans);
//         return ans;
//     }
// }